File: ExpensesTracker.Web/src/app/models/expense.ts
export type Expense = {
  id: number;
  date: string; // YYYY-MM-DD
  amount: number;
  description?: string;
  category: string;
};

export type CreateExpenseRequest = {
  date: string; // YYYY-MM-DD
  amount: number;
  description?: string;
  category: string;
};

export type UpdateExpenseRequest = CreateExpenseRequest;

File: ExpensesTracker.Api/Models/Expense.cs
namespace ExpensesTracker.Api.Models;

public class Expense
{
    public int Id { get; set; }
    public DateOnly Date { get; set; }
    public decimal Amount { get; set; }
    public string? Description { get; set; }
    public string Category { get; set; } = string.Empty;
}
